src/ai/backend/manager/models/etcd.py

```diff
@@ -0,0 +1,257 @@
+        ctx: GraphQueryContext,
+    ) -> Sequence[ContainerRegistry]:
+        log.info(
+            "ETCD.LOAD_ALL_CONTAINER_REGISTRIES (ak:{}, key:{})",
+            ctx.access_key,
+            ETCD_CONTAINER_REGISTRY_KEY,
+        )
+        raw_registries = await ctx.shared_config.etcd.get_prefix_dict(ETCD_CONTAINER_REGISTRY_KEY)
+        registries = t.Mapping(t.String, container_registry_iv).check(raw_registries)
+        return [cls.from_row(hostname, config) for hostname, config in registries.items()]
+
+    @classmethod
+    async def load_registry(cls, ctx: GraphQueryContext, hostname: str) -> ContainerRegistry:
+        log.info(
+            "ETCD.LOAD_CONTAINER_REGISTRY (ak:{}, key:{}, hostname:{})",
+            ctx.access_key,
+            ETCD_CONTAINER_REGISTRY_KEY,
+            hostname,
+        )
+
+        raw_registries = await ctx.shared_config.etcd.get_prefix_dict(ETCD_CONTAINER_REGISTRY_KEY)
+        try:
+            raw_registry = raw_registries[hostname]
+        except KeyError:
+            raise ObjectNotFound(object_name=f"registry: {hostname}")
+        registry = container_registry_iv.check(raw_registry)
+        return cls.from_row(hostname, registry)
+
+
+class ContainerRegistries(graphene.ObjectType):
+    items = graphene.List(ContainerRegistry, required=True)
+
+
+class CreateContainerRegistry(graphene.Mutation):
+    allowed_roles = (UserRole.SUPERADMIN,)
+    result = graphene.String()
+
+    class Arguments:
+        hostname = graphene.String(required=True)
+        props = CreateContainerRegistryInput(required=True)
+
+    @classmethod
+    @privileged_mutation(
+        UserRole.SUPERADMIN,
+        lambda id, **kwargs: (None, id),
+    )
+    async def mutate(
+        cls, root, info: graphene.ResolveInfo, hostname: str, props: CreateContainerRegistryInput
+    ) -> CreateContainerRegistry:
+        ctx: GraphQueryContext = info.context
+        raw_config = {
+            "": props.url,
+            "type": props.container_type,
+            "project": props.project,
+            "username": props.username,
+            "password": props.password,
+        }
+        container_registry_iv.check(raw_config)
+        config = {key: value for key, value in raw_config.items() if value is not None}
+
+        log.info(
+            "ETCD.CREATE_CONTAINER_REGISTRY (ak:{}, key: {}, hostname:{}, config:{})",
+            ctx.access_key,
+            ETCD_CONTAINER_REGISTRY_KEY,
+            hostname,
+            config,
+        )
+        updates = {}
+
+        def flatten(o):
+            for k, v in o.items():
+                if k == "":
+                    inner_prefix = f"{ETCD_CONTAINER_REGISTRY_KEY}/{hostname}"
+                else:
+                    inner_prefix = f"{ETCD_CONTAINER_REGISTRY_KEY}/{hostname}/{k}"
+                if isinstance(v, Mapping):
+                    flatten(v)
+                else:
+                    updates[inner_prefix] = v
+
+        flatten(config)
```

Comment:
The calculated `config` seems not to be used anywhere. Please clean up the code here.

Reply:
@achimnol 
`config` removes keys with `None` values and uses it as an argument for the `flatten` function to create `updates`. Otherwise, it will be stored in etcd as `'None'`.(e.g. `username: 'None'`).
https://github.com/lablup/backend.ai/blob/da28c97afa488016d6c84e15b441675503e36bf2/src/ai/backend/manager/models/etcd.py#L194

Reply:
How about this?? I used `set_if_set`
https://github.com/lablup/backend.ai/blob/e4e304234e6312e88ad28d775ed99a6fbea314de/src/ai/backend/manager/models/etcd.py#L118-L122